qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/commons/snippets/SnippetRegistry.java

```diff
@@ -303,10 +308,15 @@ private static MarkupContent createDocumentation(Snippet snippet, Map<String, St
 	}
 
 	private static String getInsertText(Snippet snippet, Map<String, String> model, boolean replace,
-			String lineDelimiter) {
+			String lineDelimiter, Node node) {
 		StringBuilder text = new StringBuilder();
 		int i = 0;
+
+		// configure completion when nested in if section
+		boolean hasParent = node.getParent().getKind() == NodeKind.Section && snippet.getParent() != null;

```

Comment:
Please dont update çommons cl1ss with qute geature. To check that see https://github.com/eclipse/lsp4mp/blob/master/microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/snippets/SnippetContextForJava.java isMatch method. You need to have this similar class but for qute